### DISCUSSION TITLE
Restricting bearer auth to route + HTTP verb?

### DISCUSSION THREAD

**Discussion by VirtualWolf** [honojs/hono#793] [Q&A] [answered]
Hi! I'm playing around with Hono on Cloudflare and it seems very slick, nice work. :) One thing I haven't been able to figure out is how to add bearer auth to a specific route _and_ a specific HTTP verb combination. So say I have an API at `/api/posts`, I want all `GET` requests to be allowed, but only someone authorised via a bearer token should be able to make a `POST` requests to that endpoint.

Many thanks!

**Comment by yusukebe** [comment, marked as answer]
Hi @VirtualWolf !

You can write like the following:

```ts
app.get('/api/posts', (c) => {
  return c.text('Read Posts!')
})

app.post('/api/post', bearerAuth({ token: 'foo' }), (c) => {
  return c.text('Post Created!', 201)
})
```

**Reply by VirtualWolf** [reply]
Oohhhh, right! Amazing, thanks for the fast response! I think this'd be good to have as an example in [the docs](https://honojs.dev/docs/builtin-middleware/bearer-auth/)? Happy to raise a PR for it. :)

**Reply by yusukebe** [reply]
> Happy to raise a PR for it.

Please :)

**Reply by VirtualWolf** [reply]
Done: https://github.com/honojs/honojs.dev/pull/29

**Reply by yusukebe** [reply]
Cool! Thanks!

**Comment by ghost** [comment]


**Reply by yusukebe** [reply]
@hclj37 

Please create another issue.

**Reply by ghost** [reply]
Sorry, I created a new one here: https://github.com/orgs/honojs/discussions/809
